File: src/main/java/com/mystchonky/machina/common/nexus/IExtendedNexusData.java
package com.mystchonky.machina.common.nexus;

import net.minecraft.nbt.CompoundTag;
import net.neoforged.neoforge.common.util.INBTSerializable;

interface IExtendedNexusData<T extends IExtendedNexusData<T>> extends INBTSerializable<CompoundTag> {
}

File: src/main/java/com/mystchonky/machina/common/nexus/IExtendedNexusData.kt
package com.mystchonky.machina.common.nexus

import net.minecraft.nbt.CompoundTag
import net.neoforged.neoforge.common.util.INBTSerializable

interface IExtendedNexusData<T : IExtendedNexusData<T>> : INBTSerializable<CompoundTag>
